ethcore/src/engines/tendermint/mod.rs

```diff
@@ -103,6 +103,7 @@ pub struct Tendermint {
 
 struct EpochVerifier {
 	subchain_validators: SimpleList,
+	recover: Box<Fn(&Signature, &Message) -> Result<Address, Error> + Send + Sync>,
```

Comment:
I think we might need a virtual here if we want to have the `signature`s used in the mock be grabbed from the `tap` provided by `setup()`. The alternative would be to hard code those addresses. I'll push a commit for this and see what you think